applications/asset_tracker/src/main.c

```diff
@@ -1501,15 +1501,15 @@ void connection_evt_handler(const struct cloud_event *const evt)
 	if (evt->type == CLOUD_EVT_CONNECTING) {
 		LOG_INF("CLOUD_EVT_CONNECTING");
 		ui_led_set_pattern(UI_CLOUD_CONNECTING);
-		k_delayed_work_cancel(&cloud_reboot_work);
+		k_work_cancel_delayable(&cloud_reboot_work);
 
 		if (evt->data.err != CLOUD_CONNECT_RES_SUCCESS) {
 			cloud_connect_error_handler(evt->data.err);
 		}
 		return;
 	} else if (evt->type == CLOUD_EVT_CONNECTED) {
 		LOG_INF("CLOUD_EVT_CONNECTED");
-		k_delayed_work_cancel(&cloud_reboot_work);
+		k_work_cancel_delayable(&cloud_reboot_work);
```

Comment:
If this fails, we'll unconditionally shut down. Perhaps the reboot work handler could check the state variable before shutting down?

Reply:
There is no good state variable to use in this case. We can live with a reboot in corner cases. More risky to add state handling to this application